client/src/elm/Backend/Update.elm

```diff
@@ -3935,76 +3952,103 @@ generatePrenatalAssessmentMsgs currentDate language isChw activePage updateAsses
                             |> Backend.Model.MsgPrenatalEncounter id
                             |> App.Model.MsgIndexedDb
 
+                    addedDiagnoses =
+                        EverySet.diff diagnosesAfter diagnosesBefore
+                            |> EverySet.toList
+
                     initialEncounterMsgs =
                         let
                             urgentDiagnoses =
-                                EverySet.toList diagnosesAfter
-                                    |> List.filter (\diagnosis -> List.member diagnosis Pages.PrenatalActivity.Utils.emergencyReferralDiagnoses)
-
-                            instructions =
-                                if
-                                    List.any
-                                        (\maternityWardDiagnosis ->
-                                            List.member maternityWardDiagnosis urgentDiagnoses
-                                        )
-                                        Pages.PrenatalActivity.Utils.maternityWardDiagnoses
-                                then
-                                    Translate.DangerSignsHelperReferToMaternityWard
+                                List.filter
+                                    (\diagnosis ->
+                                        List.member diagnosis Pages.PrenatalActivity.Utils.emergencyReferralDiagnoses
+                                    )
+                                    addedDiagnoses
 
-                                else
-                                    Translate.DangerSignsHelperReferToEmergencyObstetricCareServices
-                        in
-                        if List.isEmpty urgentDiagnoses then
-                            [ updateDiagnosesMsg ]
+                            additionalMsgs =
+                                if List.isEmpty urgentDiagnoses then
+                                    []
 
-                        else
-                            let
-                                signs =
-                                    List.map (Translate.PrenatalDiagnosis >> translate language) urgentDiagnoses
-                                        |> String.join ", "
+                                else
+                                    case addedDiagnoses of
+                                        [] ->
+                                            []
 
-                                message =
-                                    if isChw then
-                                        translate language Translate.DangerSignsLabelForChw ++ ": " ++ signs
+                                        [ DiagnosisPrescribeMebendezole ] ->
```

Comment:
@anvmn This seems an special case between `[]` and `[ first::rest ]` do we have any relevant comment to add here?

Reply:
@mariano-dagostino Right. Added comments.